### PR TITLE
feat: gnosis on mt pelerin

### PR DESCRIPTION
**Pull request by lmyslinski**
## Description

This PR adds Gnosis chain to MtPelerin on-ramp

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

- invalid asset mapping could result in assets not shown, resulting in confusion for the user

## Testing

- verify tokens listed are supported both on Shapeshift and on MtPelerin for Gnosis

## Screenshots (if applicable)

Zrzut ekranu 2023-05-26 o 14 44 51
Zrzut ekranu 2023-05-26 o 14 45 03

### CODE REVIEW

**Comment by firebomb1** [comment]
Hello, 

I think these assets should also be added in `MtPelerinSymbolToAssetIds` with the proper `caip` strings as they are supported by the ramp on gnosis... would be awesome to see Mt Pelerin offered for these asset we support on Gnosis..

: https://github.com/shapeshift/web/blob/8d85c69a44dc067010fac5715d6d3c94f4f6a3e8/packages/caip/src/adapters/mtpelerin/index.ts#L76-L81

File: packages/caip/src/adapters/mtpelerin/index.ts [hunk 1]
```diff
@@ -20,6 +22,7 @@ import {
 const MtPelerinSymbolToAssetIds: Record<string, AssetId[]> = {
   ETH: [ethAssetId, optimismAssetId],
   MATIC: [polygonAssetId],
+  XDAI: [gnosisAssetId],
   BTC: [btcAssetId],
   DAI: ['eip155:1/erc20:0x6b175474e89094c44da98b954eedeac495271d0f'],
   USDT: [
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/mtpelerin/index.ts:28]: <i class='graphite__hidden'>[Re: lines 75 to 75]<br/></i>

remove me?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/shapeshift/web/4585?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: packages/caip/src/adapters/mtpelerin/index.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ import {
   optimismChainId,
   polygonAssetId,
   polygonChainId,
+  gnosisAssetId,
+  gnosisChainId,
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/mtpelerin/index.ts:16]: `yarn lint:fix` this guy

File: packages/caip/src/adapters/mtpelerin/index.ts (hunk 1 above)
Comment (gomesalexandre) [inline packages/caip/src/adapters/mtpelerin/index.ts:28]: <i class='graphite__hidden'>[Re: lines 76 to 88]<br/></i>

Also remove the ones that are now supported after adding them<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/shapeshift/web/4585?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
> Hello,
> 
> I think these assets should also be added in `MtPelerinSymbolToAssetIds` with the proper `caip` strings under the corresponding symbols as they are supported by the ramp on gnosis... would be awesome to see Mt Pelerin offered for these assets when we support them on Gnosis.
> 
> :
> 
> https://github.com/shapeshift/web/blob/8d85c69a44dc067010fac5715d6d3c94f4f6a3e8/packages/caip/src/adapters/mtpelerin/index.ts#L76-L81

Absolutely @firebomb1, nice catch ser. Confirmed on the MtPelerin widget those are (still) supported for Gnosis Chain in Mt Pelerin, @lmyslinski let's remove the comments and add them under the respective umbrella

**Comment by lmyslinski** [comment]
EUR: 
Zrzut ekranu 2023-05-26 o 15 59 41

CHF: 
Zrzut ekranu 2023-05-26 o 15 59 35

USDT/USDC/WBTC/WETH
image

File: packages/caip/src/adapters/mtpelerin/index.ts [hunk 3]
```diff
@@ -20,31 +22,39 @@ import {
...
   ],
   jEUR: [
     'eip155:1/erc20:0x0f17bc9a994b87b5225cfb6a2cd4d667adb4f20b',
     'eip155:137/erc20:0x4e3decbb3645551b8a19f0ea1678079fcb33fb4c',
+    'eip155:100/erc20:0x9fB1d52596c44603198fB0aee434fac3a679f702',
   ],
   jCHF: [
     'eip155:1/erc20:0x53dfea0a8cc2a2a2e425e1c174bc162999723ea0',
     'eip155:137/erc20:0xbd1463f02f61676d53fd183c2b19282bff93d099',
+    'eip155:100/erc20:0x2d5563da42b06FbBF9c67b7DC073cF6A7842239e',
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/mtpelerin/index.ts:48]: jEUR and jCHF are unsupported on our side I reckon?

image

File: packages/caip/src/adapters/mtpelerin/index.ts [hunk 4]
```diff
@@ -69,13 +79,6 @@ const MtPelerinSymbolToAssetIds: Record<string, AssetId[]> = {
  * The following is the list of assets that Mt Pelerin supports
  * but they're **not supported** (yet?) in ShapeShift
  * {
-    '[xdai_mainnet]/0x0000000000000000000000000000000000000000': 'XDAI',
-    '[xdai_mainnet]/0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83': 'USDC',
-    '[xdai_mainnet]/0x4ECaBa5870353805a9F068101A40E0f32ed605C6': 'USDT',
-    '[xdai_mainnet]/0x8e5bBbb09Ed1ebdE8674Cda39A0c169401db4252': 'WBTC',
-    '[xdai_mainnet]/0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1': 'WETH',
```
Comment (gomesalexandre) [inline packages/caip/src/adapters/mtpelerin/index.ts:76]: What about this one?

image

**Comment by lmyslinski** [comment]
addet WETH, cleaned up unsupported assets

File: packages/caip/src/adapters/mtpelerin/index.ts (hunk 4 above)
Comment (lmyslinski) [reply packages/caip/src/adapters/mtpelerin/index.ts:76]: https://github.com/shapeshift/web/pull/4585/commits/de386f694771decf4d99e0fb970eee05cbed57ed

**Comment by firebomb1** [comment]
Seems like we do support jCHF and jEUR. I got fooled by the Global Search yesterday when I looked at it and reported the "bug" in Discord that the Asset search was more exhaustive than the Global Search for assets. I think they could be added too.

**Review by gomesalexandre — DISMISSED** [review]
Fixed the AssetId casing issue in 224b931e802b20a6187059d1a6c9765c7d9097ce

### Buy

image
image
image
image
image
image


### Sell - tested by monkey-patching `hasEnoughBalance` in `<FiatRampButton />`

image
image
image
image
image
image

File: packages/caip/src/adapters/mtpelerin/index.ts (hunk 3 above)
Comment (gomesalexandre) [reply packages/caip/src/adapters/mtpelerin/index.ts]: Confirmed unsupported, resolved as these have been reverted

File: packages/caip/src/adapters/mtpelerin/index.ts (hunk 1 above)
Comment (gomesalexandre) [reply packages/caip/src/adapters/mtpelerin/index.ts:28]: https://github.com/shapeshift/web/commit/1d8620b1d9723ed0bbc0f7461bf60a7c490b42e4

Comment (gomesalexandre) [reply packages/caip/src/adapters/mtpelerin/index.ts:28]: https://github.com/shapeshift/web/commit/1d8620b1d9723ed0bbc0f7461bf60a7c490b42e4

**Review by gomesalexandre — COMMENTED** [review]
> Seems like we do support jCHF and jEUR. I got fooled by the Global Search yesterday when I looked at it and reported the "bug" in Discord that the Asset search was more exhaustive than the Global Search for assets. I think they could be added too on the ramp supported assets.
> 
> image image

v. nice catch on jCHF and jEUR not being available through global search, was extremely confused as to how these were unavailable though seemingly available in the assets slice (and trying to manually access them by URL with checksummed asset reference didn't help 🧞)

added in 7ff50d382

**Review by gomesalexandre — APPROVED** [review]
Retest following 7ff50d38247331f32e1170bc807673739a9b7e7a assets.

Note, missing icons, chain symbol on the buy asset selection is an upstream issue, nothing we can do here. The URL itself looks good and does contains xdai as a chain. Ideally we'd disable on the sell side, but we don't have such concept yet for Mt Pelerin unfortunately

- jEUR buy

image

- jEUR sell

image

- jCHF buy

image

- jCHF sell

image

**Pull request merged**